future-apps/actor/test/actor.js

```diff
@@ -81,118 +82,121 @@ contract('Actor app', (accounts) => {
+        })
+
+        it('can execute actions', async () => {
+          const N = 1102
+
+          const data = executionTarget.contract.setCounter.getData(N)
+          const receipt = await actor.execute(executionTarget.address, depositAmount, data, { from: executor })
+
+          assertEvent(receipt, 'Execute')
+          assert.equal(await executionTarget.counter(), N)
+          assert.equal(await getBalance(executionTarget.address), depositAmount)
+          assert.equal(await getBalance(actor.address), 0)
+        })
+
+        it('can execute actions without data', async () => {
+          const noData = '0x'
+          const receipt = await actor.execute(executionTarget.address, depositAmount, noData, { from: executor })
+
+          assertEvent(receipt, 'Execute')
+          // Fallback just runs ExecutionTarget.execute()
+          assert.equal(await executionTarget.counter(), 1)
+          assert.equal(await getBalance(executionTarget.address), depositAmount)
+          assert.equal(await getBalance(actor.address), 0)
+        })
+
+        it('can execute cheap fallback actions', async () => {
+          const cheapFallbackTarget = await DestinationMock.new(false)
+          const noData = '0x'
+          const receipt = await actor.execute(cheapFallbackTarget.address, depositAmount, noData, { from: executor })
+
+          assertEvent(receipt, 'Execute')
+          assert.equal(await getBalance(executionTarget.address), depositAmount)
+          assert.equal(await getBalance(actor.address), 0)
+        })
+
+        it('can execute expensive fallback actions', async () => {
+          const expensiveFallbackTarget = await DestinationMock.new(true)
+          assert.equal(await expensiveFallbackTarget.counter(), 0)
+
+          const noData = '0x'
+          const receipt = await actor.execute(expensiveFallbackTarget.address, depositAmount, noData, { from: executor })
+
+          assertEvent(receipt, 'Execute')
+          // Fallback increments counter
+          assert.equal(await expensiveFallbackTarget.counter(), 1)
+          assert.equal(await getBalance(executionTarget.address), depositAmount)
+          assert.equal(await getBalance(actor.address), 0)
+        })
+
+        it('fails to execute without permissions', async () => {
+          const data = executionTarget.contract.execute.getData()
+
+          await assertRevert(() =>
+            actor.execute(executionTarget.address, depositAmount, data, { from: nonExecutor })
+          )
+        })
+
+        it('fails to execute when target is not a contract', async () => {
+          const nonContract = accounts[8] // random account
+          const randomData = '0x12345678'
+          const noData = '0x'
+
+          await assertRevert(() =>
+            actor.execute(nonContract, depositAmount, randomData, { from: executor })
+          )
+
+          await assertRevert(() =>
+            actor.execute(nonContract, depositAmount, noData, { from: executor })
```

Comment:
Why is this failing if `require(isContract(_target))` was removed?

Reply:
Good catch, fixed now in 7c68809.